hack/util.sh

```diff
@@ -295,3 +295,39 @@ function util::deploy_webhook_configuration() {
   kubectl apply -f "${temp_path}/temp.yaml"
   rm -rf "${temp_path}"
 }
+
+# util::wait_service_external_ip give a service external ip when it is ready, if not, wait until timeout
+# Parameters:
+#  - $1: service name in k8s
+#  - $2: namespace
+SERVICE_EXTERNAL_IP=""
+function util::wait_service_external_ip() {
+  local service_name=$1
+  local namespace=$2
+  local external_ip
+  for tmp_i in {1..30}; do
+    set +e
+    external_ip=$(kubectl get service "${service_name}" -n "${namespace}" --template="{{range .status.loadBalancer.ingress}}{{.ip}}{{end}}")
+    set -e
+    if [[ -z "$external_ip" ]]; then
+      echo "wait the external ip of ${service_name} ready..."
+      sleep 6
+      continue
+    else
+      for tmp_j in {1..10}; do
+        set +e
+        connect_test=$(curl -s -k -m 5 https://"${external_ip}":5443/readyz)
+        set -e
+        if [[ "${connect_test}" = "ok" ]]; then
+          SERVICE_EXTERNAL_IP="${external_ip}"
```

Comment:
I am so sorry that it can not return a non-integer in shell.  I can provide a `getLoadBalancerIP` function to echo the `external IP` but it implemented also by a global variable (not  a global environment) such as:
```
function util::wait_service_external_ip() {
  ...
  SERVICE_EXTERNAL_IP=${external_ip}
  ...
}

function util::getLoadBalancerIP(){
  echo ${SERVICE_EXTERNAL_IP}
}
```
the variable `SERVICE_EXTERNAL_IP` above also a  global variable :sweat_smile:  can we do it like this?

Reply:
I can split the `readyz`  test in `getLoadBalancerIP` at the same time